api/src/main/java/io/minio/GetObjectResponse.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio;
+
+import java.io.IOException;
+import java.io.InputStream;
+import okhttp3.Headers;
+
+public class GetObjectResponse extends InputStream {
+  private BaseResponse response;
+  private InputStream body;
+
+  public GetObjectResponse(
+      Headers headers, String bucket, String region, String object, InputStream body) {
+    this.response = new BaseResponse(headers, bucket, region, object);
+    this.body = body;
```

Comment:
`GetObjectResponse` can extend from `BufferedInputStream` and then call `super(body)` here inside the constructor.
With this, The class would authentically become an `InputStream` and would not need to store the `body` variable and write wrapper methods for it.

Reply:
1. We would need to keep proxying methods in `GenericResponse` here to comply with other Response classes; else `GetObjectResponse` looks different. The idea is `GetObjectResponse` is same as `GenericResponse` but also an `InputStream`.

2. Yes. Making `GenericResponse` as non-abstract class.

3. I am going to use `FilterInputStream` which is more appropriate than `BufferedInputStream`